crates/auth/src/models.rs

```diff
@@ -309,4 +330,66 @@ mod tests {
         assert_eq!(found_token.expires_at, token.expires_at);
         assert_eq!(found_token.is_enabled, token.is_enabled);
     }
+
+    #[test]
+    fn token_with_headers() {
```

Comment:
Since HTTP headers are case-insensitive, we should lowercase all headers that come from the request and store them as they are (https://stackoverflow.com/a/5259004). This is just my opinion. You should also add a test that tests in this kind of situation: where you add two headers, one with capital cases and the other with lowercase. But at the end of the day, you should only have one, and the one that is added later should override the old one instead of having two different headers.